text/0001-agile-coretime.md

```diff
@@ -0,0 +1,486 @@
+
+#### 5. `contribute`
+
+Regions may be consumed in exchange for a pro rata portion of the Instantaneous Coretime Sales Revenue from its period and regularity.
+
+A dispatchable `contribute(region: RegionId, beneficiary: AccountId)` SHALL be provided.
+
+#### 6. Renewals
+
+A dispatchable `renew(core: CoreIndex)` SHALL be provided. Any account may call `renew` to purchase Bulk Coretime and renew an active allocation for the given `core`.
+
+This MUST be called during the `LEADIN_PERIOD` prior to a Bulk sale (exactly like `purchase`) and has the same effect as `purchase` followed by `allocate` containing the same `ParaId`, except that:
+
+1. The purchase always succeeds and as such must be processed prior to regular `purchase` orders.
+2. The price of the purchase is the previous `price` incremented by `RENEWAL_PRICE_CAP` of itself or the regular price, whichever is lower.
+3. The Region must not have been split. (Transfer and striding are allowed.)
+4. The Region must be allocated in exactly the same way as before.
+
+#### 7. Migrations
+
+It is intended that paras which hold a lease from the legacy slot auction system are able to seamlessly transition into the Agile Coretime framework.
+
+A dispatchable `migrate(core: CoreIndex)` SHALL be provided. Any account may call `migrate` to purchase Bulk Coretime at the current price for the given `core`.
+
+This MUST be called during the `LEADIN_PERIOD` prior to a Bulk sale (exactly like `purchase`) and has the same effect as `purchase` followed by `allocate` with a value of `ParaId` containing just one item equal to `target`, except that:
+
+1. The purchase always succeeds and as such MUST be processed prior to regular `purchase` orders.
+2. There MUST be an ongoing legacy parachain lease whose parachain is assigned to `core` and which is due to expire during the Coretime period being purchased.
+
+AllowedRenewals map is updated with this information ready for the following Bulk sale.
+
+### Notes on Price Setting
+
+The specific price setting mechanism is out of scope for this proposal and should be covered in some other work. The present proposal assumes the existence of a price-setting mechanism which could take into account three parameters; two mostly fixed and one which changes each `BULK_PERIOD`. These parameters are `BULK_TARGET`, `BULK_LIMIT` and `CORES_SOLD` which is the actual number of cores sold for the present `BULK_PERIOD` and is always an unsigned integer at most `BULK_LIMIT`.
+
+In general we would expect the price to increase the closer `CORES_SOLD` gets to `BULK_LIMIT` and to decrease the closer it gets to zero. If it is exactly equal to `BULK_TARGET`, then we would expect the price to remain the same.
+
+A simple example of this would be the formula:
+
+```
+NEW_PRICE := IF CORES_OLD < BULK_TARGET THEN
+    OLD_PRICE - OLD_PRICE / 2 * CORES_SOLD / BULK_TARGET
```

Comment:
the left side of the function is flipped: it goes from OLD_PRICE @ 0 to OLD_PRICE/2 @ BULK_TARGET

The correct left side of the function looks like this:
OLD_PRICE / 2 + OLD_PRICE / 2 * CORES_SOLD / BULK_TARGET

or

OLD_PRICE / 2 * (1 + CORES_SOLD / BULK_TARGET)